packages/jest-runner/src/types.ts

```diff
@@ -37,12 +41,45 @@ export type OnTestSuccess = (
   testResult: TestResult,
 ) => Promise<void>;
 
+// Typings for `sendMessageToJest` events
+
+export type TestFileStart = {
+  eventName: 'test-file-start';
+  args: [Test];
+};
```

Comment:
```suggestion
export type TestFileStart = (eventName: 'test-file-start', test: Test) => Promise<void>`
```

same for the others

Reply:
Do I split the args for others too (like for `[ Test, TestResult ]` ? I had preferred args, so it would be easier to pass the array as a whole message to the eventListener (coz eventListener just accepts eventName and message)

Reply:
@sauravhiremath I made these changes and faced the same issue as you mentioned in the comment below. Changing this would also require changes in other function arguments.